ada/language/ast.py

```diff
@@ -359,14 +359,19 @@ def referenced_decl_internal_helper(ref_var=T.LogicVar,
         ).then(lambda x: x.match(
             # If the logic variable is bound to a GenericSubpInternal, retrieve
             # the instantiation leading to it instead.
-            lambda g=T.GenericSubpInternal: entity_no_md(
-                type=T.BasicDecl,
+            lambda g=T.GenericSubpInternal: T.BasicDecl.entity.new(
                 node=g.info.rebindings.new_env.env_node
                 .cast_or_raise(T.GenericInstantiation),
-                # Since we return the instantiation itself, remove it from
-                # its rebindings.
-                rebindings=x.info.rebindings.get_parent,
-                from_rebound=x.info.from_rebound
+                info=T.entity_info.new(
+                    # Since we return the instantiation itself, remove it from
+                    # its rebindings.
+                    rebindings=x.info.rebindings.get_parent,
+                    from_rebound=x.info.from_rebound,
+                    md=new_metadata(
+                        is_call=x.info.md.is_call,
+                        dottable_subp=x.info.md.dottable_subp
```

Comment:
I don't know if propagating dottable_subp is really relevant here, since a generic subprogram instantiation will never be dottable AFAICT.